pkg/pusher/pusher.go

```diff
@@ -190,6 +198,21 @@ LOOP:
 						err = fmt.Errorf("pusher: receipt storer address: %w", err)
 						return
 					}
+
+					po := swarm.Proximity(ch.Address().Bytes(), storerPeer.Bytes())
+					d := s.depther.NeighborhoodDepth()
+					if po < d {
+						mtx.Lock()
+						retryCounter[ch.Address().ByteString()]++
+						if retryCounter[ch.Address().ByteString()] < retryCount {
+							mtx.Unlock()
+							err = fmt.Errorf("pusher: shallow receipt depth %d, want at least %d", po, d)
+							po := swarm.Proximity(ch.Address().Bytes(), storerPeer.Bytes())
+							s.metrics.ReceiptDepth.WithLabelValues(strconv.Itoa(int(po))).Inc()
```

Comment:
I suggest to keep the metrics of retried chunks because of shallow depths separate from chunks that were final. Perhaps just a counter that keeps track of shallow retries and then another one of shallow accepted (because we exhaused our retry counts)